examples/MNIST_scikit_saliency.ipynb

```diff
@@ -0,0 +1,572 @@
+    "As is shown by this example, it is easy to visualize the decision boundaries of each class by simply plotting their respective model's coefficients in the same dimensions as the input image.\n",
+    "\n",
+    "We use the XAITK-Saliency high-level API to mimic this visualization by creating saliency maps for several images from the dataset and averaging them to create a global decision boundary representation for each class.\n",
+    "This approach achieves comparable results to those shown in scikit-learn's example while requiring zero knowledge of the intrinsic properties of the model used.\n",
+    "\n",
+    "We do the same while using the `MLPClassifier` class also from the scikit-learn library.\n",
+    "Our model is taken from [another example](https://scikit-learn.org/stable/auto_examples/neural_networks/plot_mnist_filters.html#sphx-glr-auto-examples-neural-networks-plot-mnist-filters-py) on their website.\n",
+    "They show in the example a visualization of the MLP's weights as images to gain insight on the learning behavior of the model.\n",
+    "However, these images only show generic patterns that the model has learned, however and the behavior of each class is still obscure.\n",
+    "\n",
+    "Using our approach provides per-class saliency representations and also gives some insight on the different learning behaviors portrayed by both these models.\n",
+    "\n",
+    "To run this notebook in Colab, please use the link below:\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/XAITK/xaitk-saliency/blob/master/examples/MNIST_scikit_saliency.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "265104e6",
+   "metadata": {},
+   "source": [
+    "## Setting up Environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "322e6692",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Traceback (most recent call last):\n",
+      "  File \"<string>\", line 1, in <module>\n",
+      "ModuleNotFoundError: No module named 'xaitk_saliency'\n",
+      "\u001b[33mWARNING: Discarding https://files.pythonhosted.org/packages/df/86/3db1d064d8a04fae4d4e23262a1894e27de31a490e60d21324452bac7f14/xaitk_saliency-0.3.1-1.tar.gz#sha256=a8cee265686db1422f1a0cb77b68a97c5895d168aaaa62cb14e7d21ebfcc592e (from https://pypi.org/simple/xaitk-saliency/) (requires-python:>=3.7,<3.10). Requested xaitk-saliency from https://files.pythonhosted.org/packages/df/86/3db1d064d8a04fae4d4e23262a1894e27de31a490e60d21324452bac7f14/xaitk_saliency-0.3.1-1.tar.gz#sha256=a8cee265686db1422f1a0cb77b68a97c5895d168aaaa62cb14e7d21ebfcc592e has inconsistent version: filename has '0.3.1.post1', but metadata has '0.3.1'\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "!python -c \"import xaitk_saliency\" || pip install -q \"xaitk-saliency\""
```

Comment:
This error maybe suppressed by adding a STDERR redirect in the lhs command:
```python
!python -c "import xaitk_saliency" 2>/dev/null || pip install -q "xaitk-saliency"
```

Of course, checking if it is installed is a little overkill as a simpler invocation of `!pip install -q xaitk-saliency` will be a no-op when the package is already installed.

Reply:
Do you know what is the cause of the output message? In my local poetry dev environment with jupyter notebook installed, running this cell is effectively a noop.

Reply:
The warning is the failure of the `python -c "import xaitk_saliency"` call, which is expected if the package is not installed. As mentioned above, we don't *need* this check since the pip install is idempotent.